bin/parallel.py

```diff
@@ -158,6 +152,18 @@ def join(self):
 
     # Wait for all tasks to be done and stop all threads
     def done(self):
+        if not self.num_threads:
+            # no task will be handled after self.abort
+            while self.tasks and not self.abort:
+                task = heapq.heappop(self.tasks)
+                if self.pin:
+                    cores = list(os.sched_getaffinity(0))
+                    os.sched_setaffinity(0, {cores[0]})
+                    self.f(task.task)
+                    os.sched_setaffinity(0, cores)
+                else:
+                    self.f(task.task)
+
         self.finish = True
```

Comment:
Hrmmm; so in the parallel case:
- jobs start running immediately after the first `push`
- jobs can add additional jobs
- when calling `done()`, `finish` is set to true and immediately no more jobs can be added.

But now in the non-parallel case:
- jobs start on `done`
- these jobs can add more jobs
- only after they're all done, `finish` is set to true.

Probably this is OK? At least this should be documented clearly in a comment here.

Reply:
Yes, that is indeed the intention :slightly_smiling_face: This hangs on the assumption that no jobs will be added after the call to `Parallel.done()` anyway, which I think is a fair assumption to make :stuck_out_tongue: 

Documenting this is indeed a good idea, thanks :smile: